crates/bevy_ui/src/focus.rs

```diff
@@ -166,12 +204,352 @@ pub fn ui_focus_system(
             FocusPolicy::Pass => { /* allow the next node to be hovered/clicked */ }
         }
     }
-    // reset lower nodes to None
-    for (_entity, _focus_policy, interaction, _) in moused_over_z_sorted_nodes {
-        if let Some(mut interaction) = interaction {
-            if *interaction != Interaction::None {
-                *interaction = Interaction::None;
+}
+
+trait CursorResource: Resource {
+    fn get_cursor_position(&self) -> Option<Vec2>;
+}
+
+impl CursorResource for Windows {
+    fn get_cursor_position(&self) -> Option<Vec2> {
+        self.get_primary()
+            .and_then(|window| window.cursor_position())
+    }
+}
+
+#[cfg(test)]
+mod tests {
```

Comment:
These tests look good, and we definitely need tests for this piece of code.

Reply:
Is there actually a common practice how to test systems? This TestApp approach is something I came up with in one of my own projects from the Cheatbook suggesting Direct World Access. But I didn't see many tests around in bevy itself.

Reply:
You can see how I did it in one of my external crates here: https://github.com/Leafwing-Studios/leafwing_input_manager/blob/main/tests/integration.rs